stdlib/public/core/Codable.swift

```diff
@@ -5537,33 +5537,38 @@ internal struct _DictionaryCodingKey: CodingKey {
 /// unkeyed container of alternating key-value pairs.
 @available(SwiftStdlib 5.6, *)
 public protocol CodingKeyRepresentable {
+
   @available(SwiftStdlib 5.6, *)
-  var codingKey: CodingKey { get }
+  var codingKey: any CodingKey { get }
+
   @available(SwiftStdlib 5.6, *)
-  init?<T: CodingKey>(codingKey: T)
+  init?(codingKey: some CodingKey)
 }
 
 @available(SwiftStdlib 5.6, *)
-extension RawRepresentable
-where Self: CodingKeyRepresentable, RawValue == String {
+extension RawRepresentable<String> where Self: CodingKeyRepresentable {
+
   @available(SwiftStdlib 5.6, *)
-  public var codingKey: CodingKey {
+  public var codingKey: any CodingKey {
     _DictionaryCodingKey(stringValue: rawValue)
   }
+
   @available(SwiftStdlib 5.6, *)
-  public init?<T: CodingKey>(codingKey: T) {
+  public init?(codingKey: some CodingKey) {
     self.init(rawValue: codingKey.stringValue)
   }
 }
 
 @available(SwiftStdlib 5.6, *)
-extension RawRepresentable where Self: CodingKeyRepresentable, RawValue == Int {
+extension RawRepresentable<Int> where Self: CodingKeyRepresentable {
+
   @available(SwiftStdlib 5.6, *)
-  public var codingKey: CodingKey {
+  public var codingKey: any CodingKey {
     _DictionaryCodingKey(intValue: rawValue)
   }
+
   @available(SwiftStdlib 5.6, *)
-  public init?<T: CodingKey>(codingKey: T) {
+  public init?(codingKey: some CodingKey) {
```

Comment:
Be careful with this one (and the two others of the same sort below) -- the constraints are weirdly split across the extension and the member, and rewriting them could easily end up causing a subtle ABI issue. This is exploring new language territory, and the ABI checker is unlikely to be of any help (other than flagging this as a potential break), but if the exported symbols match before/after this change, then we _may_ be okay.

Reply:
> I'm inclined to believe that api-digester has checked the mangled names. 

I'm not inclined to believe that. I don't think I've seen a list of symbols that libswiftCore is expected to export anywhere in this repository.

> However, the [macos.json](https://github.com/apple/swift/tree/main/utils/api_checker/FrameworkABIBaseline/Swift) baselines haven't been updated for 3 years, so they won't include the [CodingKeyRepresentable](https://github.com/apple/swift-evolution/blob/main/proposals/0320-codingkeyrepresentable.md) protocol.

D'oh. OK, so the ABI checker results are irrelevant, and we need to do manual checks.

Cc @nkcsgexi